File: FrontEnd/GuatEmpleos/src/app/services/applicant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Category } from '../model/Category';
import { Observable } from 'rxjs';
import { Job } from '../model/Job';

@Injectable({
  providedIn: 'root'
})
export class ApplicantService {

  readonly url: string = 'http://localhost:8080/BackendGuateEmpleos_war_exploded';

  constructor(private httpClient : HttpClient) { }


  public insertOwnCategory(categoryList : Category[], numberCui : number){
    const params = new HttpParams().set('flag', '1')
    .set('numberCui', numberCui);
    return this.httpClient.post(this.url + '/applicantServlet', categoryList, {params});
  }

  public getRecomendedJobs(numberCui : number): Observable<Job[]> {
    const params = new HttpParams().set('flag', '2').set('numberCui', numberCui);
    return this.httpClient.post<Job[]>(`${this.url}/applicantServlet`, null, { params});
  }

  public insertRequest(numberCui : number, numberJobOffert : number, postulationReason : string){
    const params = new HttpParams().set('flag', '3')
    .set('numberCui', numberCui)
    .set('numberJobOffert', numberJobOffert)
    .set('postulationReason', postulationReason);
    return this.httpClient.post(this.url + '/applicantServlet', null, {params});
  }

}

File: Backend/BackendGuateEmpleos/src/main/java/com/example/backendguateempleos/controller/ApplicantService.java
package com.example.backendguateempleos.controller;

import com.example.backendguateempleos.model.Category;
import com.example.backendguateempleos.model.Job;
import com.example.backendguateempleos.querys.QueryApplicant;

import java.util.List;

public class ApplicantService {
    private final QueryApplicant queryApplicant = new QueryApplicant();

    public boolean insertOwnCategory(List<Category> categoriesList, int cuiApplicant){
        return queryApplicant.insertCategoriesForApplicant(categoriesList, cuiApplicant);
    }

    public List<Job> getRecomendedJobs(int cui){
        return queryApplicant.recomendedJobs(cui);
    }

    public boolean insertRequest(int cui, int numberJobOffert, String postulationReason){
        return queryApplicant.insertRequest(cui, numberJobOffert, postulationReason);
    }

}
